### PR TITLE
Add support for multiple types to ServiceBrowsers

### PR DESCRIPTION
**Pull request by bdraco**
As each ServiceBrowser runs in its own thread there
is a scale problem when listening for many types.

I've tried to do this in a way that provides backwards compatibility. 
All of the existing tests pass.

ServiceBrowser can now accept a list of types
in addition to a single type.

As Home Assistant adds more discoverable services this
problem continues to grow
https://github.com/home-assistant/core/blob/dev/homeassistant/components/zeroconf/__init__.py#L213

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
     readme = f.read()
 
 version = (
-    [l for l in open(join(PROJECT_ROOT, 'zeroconf', '__init__.py')) if '__version__' in l][0]
+    [ln for ln in open(join(PROJECT_ROOT, 'zeroconf', '__init__.py')) if '__version__' in ln][0]
```
Comment (jstasiak) [inline setup.py:12]: This seems out of scope of this pull request.

Comment (bdraco) [reply setup.py:12]: The CI was failing until I fixed this since `l` is apparently no longer an allowed var name.

Comment (bdraco) [reply setup.py:12]: Opened #252 which should allow this to merge cleanly on top if I did it right.
